File: DesignPatternsDotNet/PrototypeDesignPattern/KidsBook.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace PrototypeDesignPattern
{
    class KidsBook : IBookPrototype
    {
        public KidsBook()
        {
            Console.WriteLine("A Kids book is bought");
        }
        public object Clone() {
		object aBook=null;
        aBook = (KidsBook)base.MemberwiseClone();
        Console.WriteLine("A Kids book is cloned");
		return aBook;
	}
    }
}

File: DesignPatternsJava/src/main/java/com/prototype/KidsBook.java
package com.prototype;

public class KidsBook implements IBookPrototype{
	public KidsBook(){
		System.out.println("A Kids book is bought");
	}
	public IBookPrototype makeAClone() {
		KidsBook aBook=null;
		try {
			aBook = (KidsBook)super.clone();
		} catch (CloneNotSupportedException e) {
			e.printStackTrace();
		}
		System.out.println("A Kids book is cloned");
		return aBook;
	}
}

